pcdsdevices/interface.py

```diff
@@ -890,3 +890,49 @@ def update(self, *args, name=None, current=None, **kwargs):
                            current=current, **kwargs)
         else:
             super().update(*args, name=name, current=current, **kwargs)
+
+
+class LightpathMixin(OphydObject):
+    """
+    Mix-in class that makes it easier to establish a lightpath interface.
+
+    Use this on classes that are not state positioners but would still like to
+    be used as a top-level device in lightpath.
+    """
+    SUB_STATE = 'state'
+    _default_sub = SUB_STATE
+
+    # Component names whose values are relevant for inserted/removed
+    lightpath_cpts = []
+
+    def __init_subclass__(cls, **kwargs):
+        # Magic to subscribe to the list of components
+        super().__init_subclass__(**kwargs)
+        if not cls.lightpath_cpts:
+            raise NotImplementedError('Did not implement LightpathMixin')
+        for cpt_name in cls.lightpath_cpts:
+            cpt = getattr(cls, cpt_name)
+            cpt.sub_default(cls._update_lightpath)
```

Comment:
Neat use of the decorator 👍

Reply:
I always find it handy when I need to do subscription shenanigans :+1: